templates/01_types.tpl

```diff
@@ -32,8 +35,10 @@ var (
 	{{$varNameSingular}}InsertCache = make(map[string]insertCache)
 	{{$varNameSingular}}UpdateCacheMut sync.RWMutex
 	{{$varNameSingular}}UpdateCache = make(map[string]updateCache)
+	{{if ne .DriverName "mssql"}}
```

Comment:
Upsert seems like it's available in a few forms (I don't really know T-SQL, been a -long- time since I wrote it and never did an upsert, hah). Does it not fit nicely with the existing code? Or are there other issues with it?